### PR TITLE
Ocarina of Time Source Port

### PR DESCRIPTION
**Pull request by retropieuser**
Adding in ship of harkinian support for 64 bit machines, I don't have a pi 4 or x64 to test but this should work on anything with 64bit.

Can you test and confirm it works

### CODE REVIEW

**Comment by Exarkuniv** [comment]
does the rom file need to be named something specific? i know for the mario 64 port you have to rename it, is it the same for this?

**Comment by retropieuser** [comment]
No the room can be named anything, their building functions look for the hashes

**Comment by Exarkuniv** [comment]
man, trying to find the right rom is a pain in the ass, 

for now you should add that it need to be the EU version

**Comment by Exarkuniv** [comment]
so far, i cant get it to work. 

i have right roms, ones that match the hashs, and pass the hash check site that SOH uses

and you are missing the source function in the script

File: scriptmodules/ports/soh.sh [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                echo "Copied $file to $destdir"
+                return 0
+            fi
+        fi
+    done
+
+    echo "No matching ROM file found in $romdir"
+    return 1
+}
+
```
Comment (Exarkuniv) [inline scriptmodules/ports/soh.sh:65]:
```
function sources_soh() {
    gitPullOrClone
}
```

Comment (Exarkuniv) [inline scriptmodules/ports/soh.sh:65]: function sources_soh() {
   gitPullOrClone
}

**Review by Exarkuniv — CHANGES_REQUESTED** [review]
need to add this or it will not give a option to install

**Comment by retropieuser** [comment]
I might need to play with the memory swap though, as for me just now it hangs when it gets as far as 2/867 in the building process, once it's extracted the rom. How big is the ram in your PI 5?

**Comment by Exarkuniv** [comment]
i have a 4 gig, but mine did not get that far, it would always say that it could not find the right rom, but i had the right ones per the hash in the .json file. 

ill try your new updated script

**Comment by Exarkuniv** [comment]
well at least this time it found the rom, but still failed

```
-- Linking against dependent libraries statically
CMake Error at /usr/share/cmake-3.25/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find GLEW (missing: GLEW_INCLUDE_DIRS GLEW_LIBRARIES)
Call Stack (most recent call first):
  /usr/share/cmake-3.25/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.25/Modules/FindGLEW.cmake:238 (find_package_handle_standard_args)
  libultraship/extern/CMakeLists.txt:103 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/pi/RetroPie-Setup/tmp/build/soh/build-cmake/CMakeFiles/CMakeOutput.log".
```

**Comment by Exarkuniv** [comment]
My god it took forever for this to compile. 

ok, im back on testing this, 
i think we should include 2 supported hash numbers to help people out, since there are a lot of different versions. 

i think having 
        "name": "PAL 1.1",
        "sha1": "cfbb98d392e4a9d39da8285d10cbef3974c2f012"
and 
       "name": "PAL GC (Debug)",
        "sha1": "cee6bc3c2a634b41728f2af8da54d9bf8cc14099"
or 
        "name": "PAL MQ (Debug)",
        "sha1": "cfecfdc58d650e71a200c81f033de4e6d617a9f6"

what you think?

**Comment by Exarkuniv** [comment]
ok i finally got it to build. it looks like the Vulcan build i have does not like to compile this. 
not sure if you are running one with this or not

does not working for me. 
```
Parameters: 
Executing: /opt/retropie/ports/soh/soh.sh
/opt/retropie/ports/soh/soh.sh: line 7: ./soh.elf: No such file or directory

```

**Comment by Exarkuniv** [comment]
i got it to work at the last min
ill put in my fixes monday

**Comment by retropieuser** [comment]
> i got it to work at the last min ill put in my fixes monday

My notifications appeared there was a discussion on RetroPie forums, that's good to hear.

I'm abit tied up atm, but you could add those changes to the MM source port too

**Comment by Exarkuniv** [comment]
> I'm abit tied up atm

Yes if I'm thinking correctly you are moving.
So you all good. 

I will make the changes Monday and the do the pull. 
Thanks again for working on this.

**Comment by Exarkuniv** [comment]
i have figured it all out now. 
will add the fixes once merged

**Pull request merged**